inc/Editor/BlockManagement/ConfigRegistry.php

```diff
@@ -48,84 +90,116 @@ public static function register_block( array $block_config = [] ): bool|WP_Error
-					'image_url' => $display_query->get_image_url(),
+		$queries = [];
+		$selectors = [];
+		$display_queries_to_selectors_map = [];
+
+		// Generate the selectors for the selector queries.
+		foreach ( $block_config[ self::QUERIES_KEY ] as $query_key => $query ) {
+			// Inflate the query, and add it to the queries array.
+			$query = self::inflate_query( $query );
+			$queries[ $query_key ] = $query;
+
+			$input_schema = $query->get_input_schema();
+			$output_schema = $query->get_output_schema();
+
+			// Generate the selector for the display query, and then continue on to the next query.
+			if ( in_array( $query_key, $block_config['display_queries'], true ) ) {
+				$is_collection = true === ( $output_schema['is_collection'] ?? false );
+				$has_required_variables = array_reduce(
+					array_column( $input_schema, 'required' ),
+					fn( $carry, $required ) => $carry || ( $required ?? true ),
+					false
+				);
+
+				$selectors[] = [
+					'image_url' => $query->get_image_url(),
 					'inputs' => self::map_input_variables( $input_schema ),
 					'name' => $has_required_variables ? 'Manual input' : ( $is_collection ? 'Load collection' : 'Load item' ),
-					'query_key' => self::DISPLAY_QUERY_KEY,
+					'query_key' => $query_key,
 					'type' => $has_required_variables ? 'manual-input' : 'load-without-input',
-				],
-			],
-			'title' => $block_title,
-		];
+				];
 
-		// Register "selectors" which allow the user to use a query to assist in
-		// selecting data for display by the block.
-		foreach ( $block_config[ self::SELECTION_QUERIES_KEY ] ?? [] as $selection_query ) {
-			$from_query = self::inflate_query( $selection_query['query'] );
-			$from_query_type = $selection_query['type'];
-			$to_query = $display_query;
+				$display_queries_to_selectors_map[ $query_key ][] = $query_key;
 
-			$config['queries'][ $from_query::class ] = $from_query;
+				continue;
+			}
 
-			$from_input_schema = $from_query->get_input_schema();
-			$from_output_schema = $from_query->get_output_schema();
+			// Infer the type of the query, for selector generation and to validate the non-display queries.
+			$inferred_type = self::infer_query_type( $input_schema, $output_schema );
+			if ( is_wp_error( $inferred_type ) ) {
+				return $inferred_type;
+			}
 
-			foreach ( array_keys( $to_query->get_input_schema() ) as $to ) {
-				if ( ! isset( $from_output_schema['type'][ $to ] ) ) {
-					return self::create_error( $block_title, sprintf( 'Cannot map key "%1$s" from %2$s query. The display query for this block requires a "%1$s" key as an input, but it is not present in the output schema for the %2$s query. Try adding a "%1$s" mapping to the output schema for the %2$s query.', esc_html( $to ), $from_query_type ) );
-				}
+			// If the output schema's type is not an array, then skip selector generation as that'll not work.
+			// This has been done because some output schemas have the type set to string.
+			if ( ! is_array( $output_schema['type'] ) ) {
```

Comment:
In order to figure out what selector queries work for what display queries, the output schema of the selector query has to have a type that can be iterated through. That's why this check is present.

I noticed in the tests, there were some cases where the type was set to string and that was it. That prompted me to add this in. If there is a valid case that I should consider here, let me know. I'll update line 160 accordingly.

@chriszarate  - feel like you might be able to chime in here.